packages/appium/scripts/postinstall.js

```diff
@@ -1,83 +1,162 @@
 #!/usr/bin/env node
+
 /* eslint-disable no-console, promise/prefer-await-to-then */
 
+/** @type {import('../lib/cli/extension').runExtensionCommand} */
+let runExtensionCommand;
+/** @type {import('../lib/constants').DRIVER_TYPE} */
+let DRIVER_TYPE;
+/** @type {import('../lib/constants').PLUGIN_TYPE} */
+let PLUGIN_TYPE;
+/** @type {import('../lib/extension').loadExtensions} */
+let loadExtensions;
+
+const {env, util, logger} = require('@appium/support');
+
+const ora = require('ora');
+
+const log = (message) => {
+  console.error(`[Appium] ${message}`);
+};
+
+const spinner = ora({
+  text: 'Checking Appium installation...',
+  prefixText: '[Appium]',
+}).start();
+
+try {
+  ({runExtensionCommand} = require('../build/lib/cli/extension'));
+  ({DRIVER_TYPE, PLUGIN_TYPE} = require('../build/lib/constants'));
+  ({loadExtensions} = require('../build/lib/extension'));
+  spinner.succeed('Appium installation OK');
+  // suppress logs from Appium, which mess up the script output
+  logger.getLogger('Appium').level = 'error';
+} catch (e) {
+  spinner.fail(`Could not load required module(s); has Appium been built? (${e.message})`);
+  if (process.env.CI) {
+    console.error('Detected CI environment, exiting with code 1');
+    process.exitCode = 1;
+  } else {
+    process.exitCode = 0;
+  }
+  process.exit();
+}
+
 async function main() {
+  if (await env.hasAppiumDependency()) {
+    log(`Found local Appium installation; skipping automatic install of extensions.`);
+    return;
+  }
+
   const driverEnv = process.env.npm_config_drivers;
   const pluginEnv = process.env.npm_config_plugins;
 
-  if (!driverEnv && !pluginEnv) {
-    console.log('Not auto-installing any drivers or plugins');
-    return;
-  }
+  // log(`${info} Looking for extensions to automatically install...`);
+  const spinner = ora({
+    text: 'Looking for extensions to automatically install...',
+    prefixText: '[Appium]',
+  }).start();
 
-  let extension;
-  try {
-    extension = require('../build/lib/cli/extension');
-  } catch (e) {
-    throw new Error(
-      `Could not load extension CLI file; has the project been transpiled? ` + `(${e.message})`
+  if (!driverEnv && !pluginEnv) {
+    spinner.succeed(
+      `No drivers or plugins to automatically install. If desired, set the "npm_config_drivers" or "npm_config_plugins" environment variable(s) to a comma-separated list, and they will be installed automatically with Appium.`
```

Comment:
NPM sets these env vars. What the user does is add them as flags to npm install, e.g.,

```
npm install -g appium --drivers=fake,xcuitest
```

then NPM magically turns `--drivers` into `process.env.npm_config_drivers`.

all that to say, we should probably refer to the `--drivers` and `--plugins` way of doing things because that is the intended user facing feature.

Reply:
Ah, k.